sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeCompositeContinuationImpl.java

```diff
@@ -0,0 +1,366 @@
+            }
+        } else {
+            // Add children
+            final int size = keyRanges.size();
+            for (int i = 1; i < size; i++) {
+                final PartitionKeyRange keyRange = keyRanges.get(i);
+                this.compositeContinuationTokens.add(createCompositeContinuationTokenForRange(
+                    keyRange.getMinInclusive(), keyRange.getMaxExclusive(),
+                    this.currentToken.getToken()));
+            }
+        }
+    }
+
+    private static CompositeContinuationToken createCompositeContinuationTokenForRange(
+        String minInclusive,
+        String maxExclusive,
+        String token) {
+        return new CompositeContinuationToken(
+            token,
+            new Range<>(minInclusive, maxExclusive, true, false));
+    }
+
+    private void moveToNextToken() {
+        final CompositeContinuationToken recentToken = this.compositeContinuationTokens.poll();
+        if (recentToken.getToken() != null) {
+            // Normal ReadFeed can signal termination by CT null, not NotModified
+            // Change Feed never lands here, as it always provides a CT
+            // Consider current range done, if this FeedToken contains multiple ranges due
+            // to splits,
+            // all of them need to be considered done
+            this.compositeContinuationTokens.add(recentToken);
+        }
+
+        if (this.compositeContinuationTokens.size() > 0) {
+            this.currentToken = this.compositeContinuationTokens.peek();
+        } else {
+            this.currentToken = null;
+        }
+    }
+
+    private Mono<Utils.ValueHolder<List<PartitionKeyRange>>> tryGetOverlappingRangesAsync(
+        final RxPartitionKeyRangeCache partitionKeyRangeCache, final String min, final String max,
+        final Boolean forceRefresh) {
+
+        return partitionKeyRangeCache.tryGetOverlappingRangesAsync(null, this.getContainerRid(),
+            new Range<>(min, max, false, true), forceRefresh, null);
+    }
+
+    private static CompositeContinuationToken tryParseAsCompositeContinuationToken(
+        final String providedContinuation) {
+
+        try {
+            final ObjectMapper mapper = Utils.getSimpleObjectMapper();
+
+            if (providedContinuation.trim().startsWith("[")) {
+                final List<CompositeContinuationToken> compositeContinuationTokens = Arrays
+                    .asList(mapper.readValue(providedContinuation,
+                        CompositeContinuationToken[].class));
+
+                if (compositeContinuationTokens.size() > 0) {
+                    return compositeContinuationTokens.get(0);
+                }
+
+                return null;
+            } else if (providedContinuation.trim().startsWith("{")) {
+                return mapper.readValue(providedContinuation, CompositeContinuationToken.class);
+            }
+
+            return null;
+        } catch (final IOException ioError) {
+            return null;
```

Comment:
Should we log the exception here for our own debugging purposes?

Reply:
Fixed